### PR TITLE
Dome and Aisle Lights

### PR DESCRIPTION
**Pull request by DavidKeaton**
Ref: #10471

I am hoping the power and bonus values in the JSON are proper, if not do tell.

So this was pretty much done, minus the fact there were some active cache issues causing crashes, so I sloppily left it on the backburner. Trying to clear those up, so here it is! Working afaik!

-Davek

### CODE REVIEW

File: data/json/vehicle_parts.json [hunk 1]
```diff
@@ -3974,6 +3997,2249 @@
         "location" : "on_roof",
         "flags" : ["CIRCLE_LIGHT", "ODDTURN", "FOLDABLE"],
         "breaks_into" : [
+          {"item": "steel_lump", "min": 4, "max": 6},
```
Comment (DavidKeaton) [inline data/json/vehicle_parts.json]: Welp, didn't see that it was gonna be THIS LARGE because of shifting everything...

**Comment by drbig** [comment]
Great to see this being worked on. Hopefully this patch will make https://github.com/CleverRaven/Cataclysm-DDA/pull/9476 obsolete.

**Comment by DavidKeaton** [comment]
Honestly this was p much done a while ago, but was running into a weird crash. However, active item fixes galore fixed the issue, so here it is! Dust and all!

**Comment by KA101** [comment]
Meh, I got chastised for commenting in the commit rather than the diff some time ago.  Let me know when you two have it worked out.

**Comment by DavidKeaton** [comment]
I mean as far as I am concerned, it is fine to go. I just can't full translate some JSON values into something I can understand without a reference quite yet.

**Comment by KA101** [comment]
And as for sleep aids, I find the vibrator works well for that without the addictive side effects.  ;-)

File: data/json/recipes/recipe_electronics.json [hunk 2]
```diff
@@ -300,6 +300,38 @@
...
+      [ "soldering_iron", 5 ],
+      [ "toolset", 5 ]
+    ]
+  ],
+  "components": [
+    [
+      [ "sheet_metal", 1 ]
+    ],
+    [
+      [ "lightstrip_inactive", 1 ],
```
Comment (KA101) [inline data/json/recipes/recipe_electronics.json]: Lightstrips don't have an off switch.  Probably shouldn't be used here.

**Comment by DavidKeaton** [comment]
Vibrator is great. ;-)

File: data/json/vehicle_parts.json (hunk 1 above)
Comment (KA101) [inline data/json/vehicle_parts.json]: Yep, steel lumps are way out of scale for a police light.  Chunks aren't optimal either.  I'd suggest losing both.

File: data/json/recipes/recipe_electronics.json (hunk 2 above)
Comment (DavidKeaton) [inline data/json/recipes/recipe_electronics.json]: If I may speak my train of thought...
This had occurred to me too, but the source of fuel for the light strip is typically batteries, something that has no on/off circuit logic on its own. However, the source power can have an on/off switch. This switch does not exist on the lightstrip, that is merely the "wiring" for the sheet metal, the strip and wires to provide light if provided with electricity. The current can be toggled at the controls.

ANYWAY... if you want me to change it I can. :-)
Anything for you @KA101

File: data/json/vehicle_parts.json [hunk 3]
```diff
@@ -3974,6 +3997,2250 @@
...
```
Comment (DavidKeaton) [inline data/json/vehicle_parts.json]: honestly I don't ever remember adding that. I know it says I did, bet I added the line "breaks_into" in...

File: data/json/recipes/recipe_electronics.json (hunk 2 above)
Comment (KA101) [inline data/json/recipes/recipe_electronics.json]: That'll do.  Leave it as is.

**Pull request merged**